test/replication-luatest/gh_8121_memtx_mvcc_replication_stream_test.lua

```diff
@@ -0,0 +1,57 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.cluster = cluster:new()
+    cg.master = cg.cluster:build_server({alias = 'master'})
+    cg.replica = cg.cluster:build_server(
+            {
+                alias = 'replica',
+                box_cfg = {
+                    replication = server.build_listen_uri('master'),
+                    memtx_use_mvcc_engine = true,
+                    txn_isolation = 'read-confirmed',
+                }
+            })
+    cg.cluster:add_server(cg.master)
+    cg.cluster:add_server(cg.replica)
+    cg.cluster:start()
+    cg.master:exec(function()
+        local s = box.schema.space.create('s')
+        s:create_index('pk')
+        s:insert{0}
+    end)
+    t.helpers.retrying({}, function()
+        cg.replica:assert_follows_upstream(cg.master:get_instance_id())
+    end)
```

Comment:
Better use `cg.replica:wait_for_vclock_of(cg.master)` here.
the fact that replica follows upstream doesn't prove anything here. It might follow master, but it might not have the same vclock yet.

Reply:
Changed to `cg.master:wait_for_downstream_to(cg.replica)`.